js/classes/inventory.js

```diff
@@ -46,11 +51,16 @@ class Inventory {
     }
 
     /**
-     * @returns {number} The amount of ammo the player currently has.
+     * @returns {Ammo} The ammo that was just used. None if the ammo 
      */
-    decreaseAmmo() {
-        if (this.currentAmmo.amount != "infinite") {
+    useCurrentAmmo() {
+        if (this.currentAmmo.amount <= 0) {
+            return "Empty";
```

Comment:
Returning `null` here seems more natural to me and would make your check in Slingshot simpler, although you could certainly argue that returning "Empty" is clearer. Just something to consider.